src/main/java/com/spring/taskmanger/service/TaskService.java

```diff
@@ -72,43 +76,55 @@
+		Optional<Task> optional = taskRepository.findByIdAndUserId(taskId, getCurrentUserID ());
 		if (!(optional.isPresent())) {
 			logger.error("Error while retriving tasks ");
-
-			throw new ResourceNotFoundException("UserId " + userId + "with TaskId " + taskId + "not found");
-
+			throw new ResourceNotFoundException("UserId " + getCurrentUserID () + "with TaskId " + taskId + "not found");
 		}
-
 		return ResponseEntity.ok().body(optional.get());
-
 	}
 
-	public ResponseEntity<Task> updateTask(Task task1, Long userId, Long taskId) throws ResourceNotFoundException {
-
-		if (!userRepository.existsById(userId)) {
+	public ResponseEntity<Task> updateTask(Task task1, Long taskId) throws ResourceNotFoundException {
+		/*
+		if (!userRepository.existsById(getCurrentUserID ())) {
 			logger.error("Error while Updating tasks ");
-			throw new ResourceNotFoundException("UserId " + userId + " not found");
+			throw new ResourceNotFoundException("UserId " + getCurrentUserID () + " not found");
 		}
+		 */
 
-		return taskRepository.findById(taskId).map(task -> {
+		return taskRepository.findByIdAndUserId(taskId, getCurrentUserID ()).map(task -> {
 			task.setDescription(task1.getDescription());
 			task.setCompleted(task1.isCompleted());
-			task.setUser(task1.getUser());
 			return ResponseEntity.ok().body(taskRepository.save(task));
-		}).orElseThrow(() -> new ResourceNotFoundException("taskId " + taskId + "not found"));
+		}).orElseThrow(() -> new ResourceNotFoundException("taskId " + taskId +"with current User"+ getCurrentUserID () + "not found"));
 
 	}
 
-	public ResponseEntity<?> deleteTask(Long userId, Long taskId) throws ResourceNotFoundException {
+	public ResponseEntity<?> deleteTask(Long taskId) throws ResourceNotFoundException {
 
-		return taskRepository.findByIdAndUserId(taskId, userId).map(task -> {
+		return taskRepository.findByIdAndUserId(taskId, getCurrentUserID ()).map(task -> {
 			taskRepository.delete(task);
-			return ResponseEntity.ok().build();
+			return ResponseEntity.ok().body("Task Succefully Deleted !");
 		}).orElseThrow(
-				() -> new ResourceNotFoundException("Task not found with taaskId " + taskId + " and userId " + userId));
+				() -> new ResourceNotFoundException("Task not found with taaskId " + taskId + " and userId " + getCurrentUserID ()));
+	}
+
+	private Long getCurrentUserID () {
+		UserDetails userDetails =
+				(UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+		String currentUserName = userDetails.getUsername();
+		Optional<User> user = userRepository.findByEmail(currentUserName);
+		Long userId = user.get().getId();
+		return userId;
+
+		/*
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		String currentPrincipalName = authentication.getName();
+		Optional<User> user = userRepository.findByEmail(currentPrincipalName);
+		Long userId = user.get().getId();
+		 */
```

Comment:
Okay, this will cause get the user many times, while we can pass it in the request when we do authorization in the first place and then pass it to the service.

Reply:
private Long getCurrentUserID () {
		CustomUserDetails customUserDetails =
				(CustomUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
		Long userId = customUserDetails.getId();
		return userId;
	}
this is better , right ?

Reply:
Yes, I think so